cookiecutter/hooks.py

```diff
@@ -27,17 +27,21 @@
 def valid_hook(hook_file, hook_name):
     """Determine if a hook file is valid.
 
+    Hook name must be a substring of an element in _HOOKS variable to be valid.
+
+    Hook name must be a substring of an element in _HOOKS variable to be valid.
+
     :param hook_file: The hook file to consider for validity
     :param hook_name: The hook to find
     :return: The hook file validity
     """
     filename = os.path.basename(hook_file)
     basename = os.path.splitext(filename)[0]
-    matching_hook = basename == hook_name
-    supported_hook = basename in _HOOKS
+    matching_hook = hook_name in basename
+    supported_hook = [i for i in _HOOKS if i in basename]
```

Comment:
The row above already tests that the file is a requested hook type.
It should now be enogh to test if the requested hook type is supported.
```suggestion
    supported_hook = hook_name in _HOOKS
```
Alternatively the `any()` function could be utilized to get a boolean value right away
```suggestion
    supported_hook = any(basename.startswith(supported_hook_name) for supported_hook_name in _HOOKS)
```